src/main/java/org/springframework/integration/aws/outbound/S3MessageHandler.java

```diff
@@ -312,40 +309,64 @@ private Transfer upload(Message<?> requestMessage) {
 			if (this.uploadMetadataProvider != null) {
 				this.uploadMetadataProvider.populateMetadata(metadata, requestMessage);
 			}
-			InputStream inputStream;
-			if (payload instanceof InputStream) {
-				inputStream = (InputStream) payload;
-			}
-			else if (payload instanceof File) {
-				File fileToUpload = (File) payload;
-				if (key == null) {
-					key = fileToUpload.getName();
-				}
-				try {
-					inputStream = new FileInputStream(fileToUpload);
 
+			PutObjectRequest putObjectRequest = null;
+
+			try {
+				if (payload instanceof InputStream) {
+					InputStream inputStream = (InputStream) payload;
+					Assert.state(inputStream.markSupported(),
```

Comment:
No, just commit and push is fine.
We are squashing during merge.
Please, also add your name to `@author` list for all affected classes.